app/controllers/topics_controller.rb

```diff
@@ -33,11 +34,15 @@ def create
   def update
     @topic = Topic.find(params[:id])
     authorize @topic
-    if @topic.update_attributes(params.require(:topic).permit(:name, :description, :public))
+    if @topic.update_attributes(topic_params)
       redirect_to @topic
     else
       flash[:error] = "Error saving topic. Please try again."
       render :edit
     end
   end
+  def topic_params
```

Comment:
You should add a `protected` or `private` keyword before this method.  That makes sure that this method can only be accessed from within this class.  Since this method is intended to be a helper for `create` and `update`, it wouldn't make sense to call from outside of this Controller.